src/tests/electron/common/view-controllers/app-controller.ts

```diff
@@ -87,4 +95,18 @@ export class AppController {
             'was expecting window.insightsUserConfiguration to be defined',
         );
     }
+
+    private async waitForfeatureFlagInitializer(): Promise<void> {
+        await this.client.waitUntil(
+            async () => {
+                const executeOutput = await this.client.executeAsync(done => {
+                    done((window as any).featureFlagsController != null);
```

Comment:
I tried to share this method between both window objects (featureFlagsController, insightsUserConfiguraiton) e.g. `(window as any)[property]` but the inner scope of executeAsync didn't have access to the passed in variable. Suggestions appreciated

Reply:
You need to pass along any variables you want via parameters to the inner function and arguments to `executeAsync`, like this:

```ts
await client.executeAsync((innerFoo, innerBar, done) => {
    done(functionUsing(innerFoo, innerBar));
}, foo, bar);
```

See [webdriverio's executeAsync docs](https://webdriver.io/docs/api/browser/executeAsync.html)

Reply:
Should have thought to look there, the docs are super clear. Thanks!